plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -494,67 +519,94 @@ applyBuiltin ctx bn args = do
   result <- dischargeError $ applyTypeSchemed bn sch f exF args
   returnCek ctx result
 
--- | Evaluate a term using the CEK machine and keep track of costing.
+-- | Evaluate a term using the CEK machine and keep track of costing, logging is optional.
 runCek
     :: ( GShow uni, GEq uni, Closed uni, uni `Everywhere` ExMemoryUsage
        , Hashable fun, Ix fun, ExMemoryUsage fun
        )
     => BuiltinsRuntime fun (CekValue uni fun)
     -> ExBudgetMode
+    -> Bool
     -> Term Name uni fun ()
-    -> (Either (CekEvaluationException uni fun) (Term Name uni fun ()), CekExBudgetState fun)
-runCek runtime mode term =
-    runCekM (CekEnv runtime mode)
-            (ExBudgetState mempty mempty)
-        $ do
-            spendBudget BAST (ExBudget 0 (termAnn memTerm))
-            computeCek [] mempty memTerm
-    where
-        memTerm = withMemory term
-
--- | Evaluate a term using the CEK machine in the 'Counting' mode.
-runCekCounting
+    -> (Either (CekEvaluationException uni fun) (Term Name uni fun ()), CekExBudgetState fun, [String])
+runCek runtime mode emitting term =
+    runCekM runtime mode emitting (ExBudgetState mempty mempty) $ do
+        spendBudget BAST (ExBudget 0 (termAnn memTerm))
+        computeCek [] mempty memTerm
+  where
+    memTerm = withMemory term
+
+-- | Evaluate a term using the CEK machine with logging disabled and keep track of costing.
+runCekNoEmit
     :: ( GShow uni, GEq uni, Closed uni, uni `Everywhere` ExMemoryUsage
        , Hashable fun, Ix fun, ExMemoryUsage fun
        )
     => BuiltinsRuntime fun (CekValue uni fun)
+    -> ExBudgetMode
     -> Term Name uni fun ()
     -> (Either (CekEvaluationException uni fun) (Term Name uni fun ()), CekExBudgetState fun)
-runCekCounting runtime = runCek runtime Counting
+runCekNoEmit runtime mode term =
+    case runCek runtime mode False term of
+        (errOrRes, s', _) -> (errOrRes, s')
 
--- | Evaluate a term using the CEK machine.
+-- | Unsafely evaluate a term using the CEK machine with logging disabled and keep track of costing.
+-- May throw a 'CekMachineException'.
+unsafeRunCekNoEmit
+    :: ( GShow uni, GEq uni, Typeable uni
+       , Closed uni, uni `EverywhereAll` '[ExMemoryUsage, PrettyConst]
+       , Hashable fun, Ix fun, Pretty fun, Typeable fun, ExMemoryUsage fun
+       )
+    => BuiltinsRuntime fun (CekValue uni fun)
+    -> ExBudgetMode
+    -> Term Name uni fun ()
+    -> (EvaluationResult (Term Name uni fun ()), CekExBudgetState fun)
+unsafeRunCekNoEmit runtime mode = first unsafeExtractEvaluationResult . runCekNoEmit runtime mode
+
+-- | Evaluate a term using the CEK machine with logging enabled.
 evaluateCek
```

Comment:
I've now added a `Note` (but haven't committed it yet):

```
{- Note [CEK runners naming convention]
A function whose name ends in @NoEmit@ does not perform logging and so does not return any logs.
A function whose name starts with @unsafe@ throws exceptions instead of returning them purely.
A function from the @runCek@ family takes an 'ExBudgetMode' parameter and returns the final
'CekExBudgetState' (and possibly logs).
A function from the @evaluateCek@ family does not return the final 'ExBudgetMode', nor does it
allow one to specify an 'ExBudgetMode'. I.e. such functions are only for fully evaluating programs
(and possibly returning logs).
-}
```

Reply:
Thanks, that's very helpful.

Reply:
Yeah, I would prefer to cut down on these if we could.

Reply:
> Yeah, I would prefer to cut down on these if we could.

I would prefer that as well, but 

- they all are useful
- naming is fairly coherent
- six functions is not that many

So I think it's not too bad.

Reply:
(I can't count, it's seven, not six, and it's an odd number 'cause `unsafeRunCek` is missing. I'll try to unify `runCek*` with `evaluateCek*`)